exchange/bitswap/message/message.go

```diff
@@ -51,18 +48,18 @@ type Exportable interface {
 type impl struct {
 	full     bool
 	wantlist map[u.Key]Entry
-	blocks   map[u.Key]*blocks.Block // map to detect duplicates
+	blocks   map[u.Key]*blocks.Block
 }
 
-func New() BitSwapMessage {
-	return newMsg()
+func New(full bool) BitSwapMessage {
```

Comment:
it's hard to discern what `message.New(true)` means elsewhere in the code.

if you want to keep it as an initalizer (which means messages cannot be reused) maybe:

```go
func NewFull() BitswapMessage { return newMsg(true) }
func NewPartial() BitswapMessage { return newMsg(false) }
```